src/main/scala/org/apache/pekko/persistence/dynamodb/journal/DynamoDBHelper.scala

```diff
@@ -32,6 +32,35 @@ import scala.concurrent.duration._
 case class LatencyReport(nanos: Long, retries: Int)
 private class RetryStateHolder(var retries: Int = 10, var backoff: FiniteDuration = 1.millis)
 
+/**
+ * Auxiliary object to help determining whether we should retry on a certain throwable.
+ */
+object DynamoRetriableException {
+  def unapply(ex: AmazonServiceException) = {
+    ex match {
```

Comment:
One thing I would add is to the exception retry  prediate is that we shouldn't retry on **ANY** 4xx error (which may already include ones in the current implementation) as retrying on 4xx errors would never succeed (unless Amazon maldesigned their API).

Reply:
4XX generally not retriable but there may some exceptions like:
* 408 Request Timeout
* 429 Too Many Requests

Reply:
Well, if you look at the docs (posted link), it states they send out pure 400 and it's OK to retry given errors.

Reply:
Thanks for clarifying, I will resolve this.